### DISCUSSION TITLE
[Question] Access Areabrick / Block Data of Document in PHP

### DISCUSSION THREAD

**Discussion by zangab** [pimcore/pimcore#7780] [Q&A / Support]
# Question
We want to build a JSON-API and we do use Pimcore in the company. However, the docs are - well - not always up-to-date and I can't find any info on that.

I want to get the data of the areabricks that are saved to a document. Currently I see a `getElements()` function but this returns just the plain data that is stored in the database. What I kinda expect is a function that also transforms this data.

So for example, this is returned by calling `$document->getElements()`:

```
[
  "headline" => "Hello my friend",
  "contentblock" => Block^ {#4144 ▶}
  "contentblock:1.answer" => Wysiwyg^ {#4147 ▶}
  "contentblock:1.question" => Input^ {#4148 ▶}
  "contentblock:2.answer" => Wysiwyg^ {#4149 ▶}
  "contentblock:2.question" => Input^ {#4150 ▶}
  "contentblock:3.answer" => Wysiwyg^ {#4151 ▶}
  "contentblock:3.question" => Input^ {#4152 ▶} 
]
```
When calling `$contentblock->getElements()`(the Block element) I just get the indices.

What I look for would be something like this:

```
[
  "headline" => "Hello my friend",
  "contentblock" => [
    [
      "type" => "my-areablock-type-key",  // in case of an areabrick
      "answer" => Wysiwyg^ {#4147 ▶},
      "input" => Input^ {#4148 ▶}
   ],
   [
      "answer" => Wysiwyg^ {#4147 ▶},
      "input" => Input^ {#4148 ▶}
   ],
   ...
]
```

Does a helper like this exist or do I have to reformat that myself?

**Comment by NiklasBr** [comment]
I believe DataHub could help here, there are _some_ docs available on fetching [Pages & editables](https://github.com/pimcore/data-hub/blob/master/doc/10_GraphQL/04_Query/01_Document_Queries.md)
